### PR TITLE
New device, folder prompts (fixes #120, fixes #330)

### LINKED ISSUE #120: Make initial node id exchange easier
Not sure how yet.

### LINKED ISSUE #330: Make initial repo setup easier
When two nodes connect and one has more repos than the other, do something nice like offer to add the repos where they are missing.

### PR DESCRIPTION
**Pull request by AudriusButkevicius**

### CODE REVIEW

**Comment by calmh** [comment]
Looks nice. :heart: Although I think the steps for ignoring unshared folders is probably unnecessary - i.e. there's really no good reason to run with a folder mismatch between two devices, either add it on one or remove it on the other?

File: gui/index.html [hunk 1]
```diff
@@ -308,6 +308,72 @@ <h3 class="panel-title">
...
+              <span translate translate-value-device="{{ device }}" translate-value-address="{{ event.data.address }}">
+                Device {%device%} ({%address%}) wants to connect. Add new device?
+              <span>
+            </p>
+          </div>
+          <div class="panel-footer clearfix">
+            <div class="pull-right">
+              <button class="btn btn-sm btn-default" ng-click="addNewDeviceID(device)"><span class="glyphicon glyphicon-ok"></span>&emsp;<span translate>Add</span></button>
+              <button class="btn btn-sm btn-danger" ng-click="ignoreRejectedDevice(device)"><span class="glyphicon glyphicon-remove"></span>&emsp;<span translate>Ignore</span></button>
+              <button class="btn btn-sm" ng-click="dismissDeviceRejection(device)"><span class="glyphicon glyphicon-time"></span>&emsp;<span translate>Later</span></button>
```
Comment (calmh) [inline gui/index.html]: Button needs a class, or renders as grey and with wrong margins. Suggest `btn-default` here and either `btn-primary` or `btn-success` for the "Add" button.

File: gui/index.html [hunk 2]
```diff
@@ -308,6 +308,72 @@ <h3 class="panel-title">
...
+               <button class="pull-right btn btn-sm btn-default" ng-click="addFolderAndShare(event.data.folder, event.data.device)" ng-if="!folders[event.data.folder]">
+                <span class="glyphicon glyphicon-ok"></span>&emsp;<span translate>Add</span>
+              </button>
+              <button class="btn btn-sm btn-default" ng-click="shareFolderWithDevice(event.data.folder, event.data.device)" ng-if="folders[event.data.folder]">
+                <span class="glyphicon glyphicon-ok"></span>&emsp;<span translate>Share</span>
+              </button>
+              <button class="btn btn-sm btn-danger" ng-click="ignoreRejectedFolder(event.data.folder, event.data.device)">
+                <span class="glyphicon glyphicon-remove"></span>&emsp;<span translate>Ignore</span>
+              </button>
+              <button class="btn btn-sm" ng-click="dismissFolderRejection(event.data.folder, event.data.device)">
```
Comment (calmh) [inline gui/index.html]: Button class as above...

File: internal/config/wrapper.go [hunk 3]
```diff
@@ -245,6 +245,32 @@ func (w *Wrapper) InvalidateFolder(id string, err string) {
...
+	for _, device := range w.cfg.IgnoredDevices {
+		if device == id {
+			return true
+		}
+	}
+	return false
+}
+
+// Returns whether or not connection attempts from the given device should be
+// silently ignored.
```
Comment (calmh) [inline internal/config/wrapper.go]: Comment is a lie. :) (But as above, I don't think we need to do this anyhow.)

**Comment by AudriusButkevicius** [comment]
Are you suggesting not to have IgnoredFolder all together? And leave only Add/Share, Later options?

**Comment by calmh** [comment]
Yep.

**Comment by calmh** [comment]
The reasoning being that devices should be ignorable since they might be old/rogue/hostile, but the folder thing is just a misconfiguration between two (presumably) cooperating devices, that should be fixed on either side.

**Comment by AudriusButkevicius** [comment]
Yeah, it now makes sense to me.

**Comment by calmh** [comment]
At some point, maybe a "Clear ignored devices & folders" option somewhere, in the hamburger meny or so. Not essential today.

**Comment by AudriusButkevicius** [comment]
Well folders won't be ignored.
But yeah, waiting for a miracle to happen... that is having more space in the settings dialog to expose more options.

**Comment by AudriusButkevicius** [comment]
Changed as per your suggestions.
Still not sure where to add the button.

**Pull request merged**